TMS.Net07.Homework.Calculator/Hard.cs

```diff
@@ -0,0 +1,138 @@
+            {
+                string[] partsOfExpression = expression.Split(Convert.ToChar(operation));
+                if (partsOfExpression.Length == 2)
+                {
+                    double number1;
+                    double number2;
+                    bool isNumber1 = double.TryParse(partsOfExpression[0], out number1);
+                    bool isNumber2 = double.TryParse(partsOfExpression[1], out number2);
+                    if (isNumber1 && isNumber2 && operation == "+")
+                    {
+                        Console.WriteLine($"{number1} {operation} {number2} = {number1 + number2}");
+                    }
+                    else if (isNumber1 && isNumber2 && operation == "-")
+                    {
+                        Console.WriteLine($"{number1} {operation} {number2} = {number1 - number2}");
+                    }
+                    else if (isNumber1 && isNumber2 && operation == "*")
+                    {
+                        Console.WriteLine($"{number1} {operation} {number2} = {number1 * number2}");
+                    }
+                    else if (isNumber1 && isNumber2 && operation == "/")
+                    {
+                        Console.WriteLine($"{number1} {operation} {number2} = {number1 / number2}");
+                    }
+                    else if (isNumber1 && isNumber2 && operation == "%")
+                    {
+                        Console.WriteLine($"{number1} {operation} {number2} = {number1 % number2}");
+                    }
+                    else if(isNumber1 && isNumber2 && operation == "p")
+                    {
+                         Console.WriteLine($"{number1} pow {number2} = {Math.Pow(number1, number2)}");
+                    }
+                    else
+                    {
+                        Console.WriteLine("Wrong expression");
+                    }
+                }
+                else
+                {
+                    Console.WriteLine("Wrong expression");
+                }
+            }
+
+            else if(operation == "sqr")
+            {
+                string[] partsOfExpression = expression.Split(Convert.ToChar('r'));
+                if (partsOfExpression.Length == 2)
+                {
+                    int number;
+                    bool isNumber = int.TryParse(partsOfExpression[1], out number);
+                    if(isNumber)
+                    {
+                        Console.WriteLine($"sqr {number} = {number * number}");
+                    }
+                    else
+                    {
+                        Console.WriteLine("Wrong expression");
+                    }
+                }
+                else
+                {
+                    Console.WriteLine("Wrong expression");
+                }
+            }
+            
+            else if(operation == "sqrt")
+            {
+                string[] partsOfExpression = expression.Split(Convert.ToChar('t'));
```

Comment:
```suggestion
                string[] partsOfExpression = expression.Split('t');
```
`'t'` is already `char`. Why should we convert char to char?

Reply:
just forgot to fix what I wrote earlier, corrected